pkg/cache/clusterqueue.go

```diff
@@ -52,18 +53,19 @@ type ResourceStats map[corev1.ResourceName]*QuotaStats
 // ClusterQueue is the internal implementation of kueue.ClusterQueue that
 // holds admitted workloads.
 type ClusterQueue struct {
-	Name              string
-	Cohort            *Cohort
-	ResourceGroups    []ResourceGroup
-	RGByResource      map[corev1.ResourceName]*ResourceGroup
-	Usage             FlavorResourceQuantities
-	Workloads         map[string]*workload.Info
-	WorkloadsNotReady sets.Set[string]
-	NamespaceSelector labels.Selector
-	Preemption        kueue.ClusterQueuePreemption
-	FlavorFungibility kueue.FlavorFungibility
-	AdmissionChecks   sets.Set[string]
-	Status            metrics.ClusterQueueStatus
+	Name                   string
+	Cohort                 *Cohort
+	ResourceGroups         []ResourceGroup
+	RGByResource           map[corev1.ResourceName]*ResourceGroup
+	Usage                  FlavorResourceQuantities
+	Workloads              map[string]*workload.Info
+	WorkloadsNotReady      sets.Set[string]
+	NamespaceSelector      labels.Selector
+	Preemption             kueue.ClusterQueuePreemption
+	FlavorFungibility      kueue.FlavorFungibility
+	AdmissionChecks        sets.Set[string]
+	AdmissionCheckStrategy kueue.AdmissionChecksStrategy
```

Comment:
maybe it's better to fuse these fields. So we only process the API once in `ClusterQueue.update`, translate everything into admissionChecksStrategy (or a map based version of it) and the rest of the code just works with the strategies.

Reply:
Fixed